Adding an ADDB user guide

faradawn opened the pull request:
Signed-off-by: faradawn <faradawny@gmail.com>

# Problem Statement
- cortx-motr lacks a thorough documentation on ADDB

# Design
- Created a ADDB guide for users to build motr, perform an IO, and construct a ADDB graph.
- No coding is involved

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide


-----
[View rendered doc/ADDB-User-Guide.md](https://github.com/faradawn/cortx-motr/blob/doc/ADDB-User-Guide/doc/ADDB-User-Guide.md)

rkothiya commented:
Can one of the admins verify this patch?

andriytk commented on doc/ADDB-User-Guide.md:
```diff
@@ -0,0 +1,345 @@
+# How to use ADDB
+- Can try it on jupyter notebook on [Chameleon Trovi](https://chameleoncloud.org/experiment/share/84c1d598-699b-4e8d-bee5-6fb7f5937504)
+- Or on a Skylake CENTOS 7.9 baremetal
```
Why is this needed? Is ADDB specific or depents on these platforms? Not sure. Please remove, if not.

andriytk commented on doc/ADDB-User-Guide.md:
```diff
@@ -0,0 +1,345 @@
+cd /mnt/extra
+git clone --recursive https://github.com/Seagate/cortx-motr.git
+cd /mnt/extra/cortx-motr
+git checkout 58f3d69e51a049eb5d5e2576040cc5ee732e2410
```
Why specific commit id? Won't it work on the latest main branch?

andriytk commented on doc/ADDB-User-Guide.md:
```diff
@@ -0,0 +1,345 @@
+- Can try it on jupyter notebook on [Chameleon Trovi](https://chameleoncloud.org/experiment/share/84c1d598-699b-4e8d-bee5-6fb7f5937504)
+- Or on a Skylake CENTOS 7.9 baremetal
+
+## Part 1 - Building Motr
```
Building motr is covered at https://github.com/Seagate/cortx-motr/blob/main/doc/Quick-Start-Guide.rst, why is this needed here? Are there some special steps needed for ADDB?

andriytk reviewed (changes requested):
More than half of this document is about how to build and start Motr. But there are such documents already (see [QSG](https://github.com/Seagate/cortx-motr/blob/main/doc/Quick-Start-Guide.rst), for example), there is no need to duplicate them.

It would be nice to have some introduction words about ADDB: what it is, why it is needed and how it can help. Providing some real-world usage examples and problem(s) analysis would be very beneficial for this document.

faradawn commented:
Hi Andriy,

Have carefully edited the ADDB_User_Guide according to your comments:
- added a description of what ADDB can be used for
- explained what this guide does: generate a request graph for bottleneck analysis
- removed the redundant motr installation steps 
- simplified the ADDB graph generating process

Thank you for taking a look at the guide. Your suggestions meant a lot!

Thanks,
Faradawn

siningwuseagate commented on doc/ADDB-User-Guide.md:
```diff
@@ -0,0 +1,79 @@
+# ADDB User Guide
+[ADDB](https://github.com/Seagate/cortx-motr/blob/main/doc/ADDB.md) can track how long each step takes during a Motr operation, therefore can furnish valuable information on what procedures serve as the bottleneck of the system. This guide provides an example on how to use ADDB to generate a request timeline graph to analyze an I/O process.
```
If I understand it right, Motr has built-in ADDB data points to record timestamps at critical points of an operation execution, ADDB doesn't track the duration of each step directly, the duration of each step is actually calculated by analysing tool such as the one mentioned in this document.

siningwuseagate commented on doc/ADDB-User-Guide.md:
```diff
@@ -0,0 +1,79 @@
+# ADDB User Guide
```
From the content of this document, looks like it is not a guide on how to use ADDB to instrument Motr or its applications, or a guide on how to parse and analyse ADDB records. IMO, something like "How-to analyse IO Operation using ADDB" may be a better title.

faradawn commented:
Dear @siningwuseagate,

I agree with your suggestions wholeheartedly. Here are the changes made accordingly:
- changed the title to "How-to analyse IO Operation using ADDB"
- refined the description to "ADDB records the activities of Motr system"

Thanks,
Faradawn

siningwuseagate commented:
> Dear @siningwuseagate,
> 
> I agree with your suggestions wholeheartedly. Here are the changes made accordingly:
> 
> * changed the title to "How-to analyse IO Operation using ADDB"
> * refined the description to "ADDB records the activities of Motr system"
> 
> Thanks, Faradawn

Thanks Faradawn,

The changes look good to me. We think this doc is a very good starting point for a comprehensive ADDB guide and we are looking to expand and improve this ADDB doc, to reflect this, could you please add WORK IN PROGRESS in the title?

faradawn commented:
Hi @siningwuseagate,

Have added "WORK IN PROGRESS" in the document title -- see if that is okay?

Thanks,
Faradawn

nikitadanilov commented on doc/ADDB_Record_Analyzing.md:
```diff
@@ -0,0 +1,79 @@
+
+## Contributors
+- Daniar Kurniawan, University of Chicago, daniar@uchicago.edu
+- Faradawn Yang, University of Chicago, faradawn@uchicago.edu
```
Are you planning to extend this to a larger document? As is, this is hardly a user guide. Also, we already have doc/addb2-primer, perhaps you should use it as a base?

faradawn commented:
Hi @nikitadanilov,

I think we don't have more content to expand this to a larger document. Have added a "based on ADDB2-primer", is that okay?

Thanks,
Faradawn

r-wambui commented:
Hi @faradawn could you please update your branch, @siningwuseagate could you please merge the PR.

faradawn commented:
Hi @r-wambui and @siningwuseagate,

Have updated the branch! 

If there is anything I can do, please let me know!

Thanks,
Faradawn

shailesh-vaidya commented:
Closing as an obsolete
